src/ui/public/kuery/ast/kql.peg

```diff
@@ -1,123 +1,150 @@
-/*
- * Kuery parser
- */
-
-/*
- * Initialization block
- */
+// Initialization block
 {
-  var nodeTypes = options.helpers.nodeTypes;
+  var { nodeTypes: { function: { buildNode }}} = options.helpers;
+}
+
+start
+  = OrQuery
+  / EmptyQuery
 
-  if (options.includeMetadata === undefined) {
-    options.includeMetadata = true;
+EmptyQuery
+  = Space* {
+    return buildNode('and', []);
   }
 
-  function addMeta(source, text, location) {
-    if (options.includeMetadata) {
-      return Object.assign(
-        {},
-        source,
-        {
-          text: text,
-          location: simpleLocation(location),
-        }
-      );
-    }
+OrQuery
+  = head:AndQuery Or tail:OrQuery {
+    return buildNode('or', [head, tail]);
```

Comment:
head/tail seems like sort of weird terminology to me here. I usually think of a head being the first item in an array, and the tail being the remaining (0 or more) items, but in this case it's always a pair, which is why I used left/right before.

Reply:
162a497